### ISSUE TITLE
[Refactoring] Add back middlewares that are registered by WebApplication

### ISSUE THREAD

**Issue by captainsafia** [dotnet/aspnetcore#45222]
## Proposed Analyzer

For middlesware that are registered automatically by the WebApplication, consider a refactoring to manually add them back to the application.

### Category

- [ ] Design
- [ ] Documentation
- [ ] Globalization
- [ ] Interoperability
- [ ] Maintainability
- [ ] Naming
- [ ] Performance
- [ ] Reliability
- [ ] Security
- [ ] Style
- [ ] Usage

### Severity Level

- [ ] Error
- [ ] Warning
- [ ] Info
- [ ] Hidden

**Comment by Tratcher** [comment]
Why?

**Comment by captainsafia** [comment]
> Why?

We were discussing the fact that the `WebApplicationBuilder` is opinionated about the order in which middlesware are inserted. This suggestion came up as a way to provide users with a refactoring that would explicitly register the middlesware that we auto-register so that users could re-adjust the ordering when needed.

**Comment by Tratcher** [comment]
Ah, hopefully thus preventing WebApplicationBuilder from adding duplicates?

**Comment by captainsafia** [comment]
> Ah, hopefully thus preventing WebApplicationBuilder from adding duplicates?

Yes. The WebApplicationBuilder actually already avoids adding duplicates so when the user does the registration in their app code in whatever order they desire we won't register the same middlewares.

**Comment by amcasey** [comment]
There's a fine line between sub-warning diagnostics and refactorings.  The key functional differences are that code-fixes can be applied in multiple places at once (i.e. fix-all) and refactorings can take user-specified spans as inputs (e.g. for extract method).

There's probably also a (relatively minor) perf consideration - analyzers everywhere even though most hidden diagnostics will never be "fixed" and refactoring run every time the cursor is moved.

Either is probably fine in this case (unless this is the sort of thing you'd want to apply to multiple locations in the same project).
